### PR TITLE
Income controller

### PR DESCRIPTION
**Pull request by ifteakhor**
- Add Income controller

I will add Model and DTOs too, but later...

### CODE REVIEW

File: src/Controllers/IncomeController.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace Server.Controllers;
+
+[ApiController]
```
Comment (veyxov) [inline src/Controllers/IncomeController.cs]: This attribute is already applied in the `ApiControllerBase`

File: src/Controllers/IncomeController.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+[Route("api/[controller]")]
+public class IncomeController : ApiControllerBase
+{
+    [HttpGet("{id}")]
+    public IActionResult Show(int id)
```
Comment (veyxov) [inline src/Controllers/IncomeController.cs]: Maybe `GetIncomeInfoById`? Explicit is better than implicit most of the times.

File: src/Controllers/IncomeController.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    {
+        return Ok();
```
Comment (veyxov) [inline src/Controllers/IncomeController.cs]: please add `throw new NotImplementedException()` in code parts that are not ready or draft.

File: src/Models/Income.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+namespace server.Models;
+
+public class Income
+{
+    public float Amount { get; set; }
```
Comment (veyxov) [inline src/Models/Income.cs]: I decided to use the `decimal` type in the project, can elaborate on the choice if needed.

File: src/Models/Income.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+    public required string Date { get; set; }
```
Comment (veyxov) [inline src/Models/Income.cs]: use `DateTime` as the type for dates with time (hour, minutes) and `DateOnly` for dates.

File: src/Models/Income.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    
+    public string? Comment { get; set; }
+
+    public required string Method { get; set; }
```
Comment (veyxov) [inline src/Models/Income.cs:11]: extract `Method` to a rich model, to make it clear what method means or add a comment, I recommend the first solution.

**Review by veyxov — CHANGES_REQUESTED** [review]

File: src/Models/Income.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+namespace server.Models;
+
+public class Income
```
Comment (veyxov) [inline src/Models/Income.cs]: Database Entities should have an `id` property, you can create a `BaseEntity` abstract class which contains that property and is inherited by all database models. This also can be used as an interface for objects that have an `id`.

File: src/Controllers/IncomeController.cs (hunk 1 above)
Comment (ifteakhor) [reply src/Controllers/IncomeController.cs]: fixed

File: src/Controllers/IncomeController.cs (hunk 3 above)
Comment (ifteakhor) [reply src/Controllers/IncomeController.cs]: added

File: src/Models/Income.cs (hunk 6 above)
Comment (ifteakhor) [reply src/Models/Income.cs:11]: what do you mean by reach model ?

File: src/Models/Income.cs (hunk 4 above)
Comment (ifteakhor) [reply src/Models/Income.cs]: What is the difference between them? But for money it's better not to use decimal/float numbers.

File: src/Models/BaseEntity.cs [hunk 8]
```diff
@@ -0,0 +1,6 @@
+namespace server.Models;
+
+public abstract class BaseEntity
+{
+    protected long Id { get; set; }
```
Comment (veyxov) [inline src/Models/BaseEntity.cs]: why long?

Comment (veyxov) [reply src/Models/BaseEntity.cs]: Also should be public for the ORM to work with it.

**Review by veyxov — CHANGES_REQUESTED** [review]

File: src/Models/BaseEntity.cs (hunk 8 above)
Comment (ifteakhor) [reply src/Models/BaseEntity.cs]: I assumed that we use `BigInteger` for id in Database

**Review by veyxov — APPROVED** [review]

**Pull request merged**